kivy/uix/recyclegridlayout.py

```diff
@@ -205,26 +206,35 @@ def get_view_index_at(self, pos):
                     break
                 iy += 1
 
-        # gridlayout counts from left to right, top to down
-        iy = len(rows) - iy - 1
-        return iy * len(cols) + ix
+        ori = self.orientation
+        if 'rl' in ori:
+            ix = len(cols) - ix - 1
+        if 'tb' in ori:
+            iy = len(rows) - iy - 1
+        return (iy * len(cols) + ix) if self._fills_row_first else \
+            (ix * len(rows) + iy)
 
     def compute_visible_views(self, data, viewport):
         if self._cols_pos is None:
             return []
         x, y, w, h = viewport
-        # gridlayout counts from left to right, top to down
+        right = x + w
+        top = y + h
         at_idx = self.get_view_index_at
-        bl = at_idx((x, y))
-        br = at_idx((x + w, y))
-        tl = at_idx((x, y + h))
-        n = len(data)
+        # 'tl' is not actually 'top-left' unless 'orientation' is 'lr-tb'.
+        # But we can pretend it is. Same for 'bl' and 'br'.
+        tl, __, bl, br = sorted((
+            at_idx((x, y)),
+            at_idx((right, y)),
+            at_idx((x, top)),
+            at_idx((right, top)),
+        ))
 
+        n = len(data)
         indices = []
-        row = len(self._cols)
-        if row:
+        stride = len(self._cols) if self._fills_row_first else len(self._rows)
+        if stride:
```

Comment:
Not sure why this truthy check is necessary. Both `len(self._cols)` and `len(self._rows)` look always greater than zero because of:

https://github.com/kivy/kivy/blob/53290db64b95a655f09e7c737b8e97fda7264451/kivy/uix/gridlayout.py#L303-L304